xcdat/temporal.py

```diff
@@ -919,10 +919,28 @@ def _group_average(self, data_var: xr.DataArray) -> xr.DataArray:
 
         if self._weighted:
             self._weights = self._get_weights()
+            # grab metadata
+            dv_attrs = dv.attrs
+            dv_name = dv.name
```

Comment:
When I updated `dv = self._group_data(dv).sum()` to `dv = self._group_data(dv).sum() / self._group_data(weights).sum()` these attributes were dropped, so I held onto them to put them back on.